src/Models/Permission.php

```diff
@@ -147,6 +139,39 @@ public static function findOrCreate(string $name, $guardName = null): Permission
         return $permission;
     }
 
+    /**
+     *  Find a permission by its name or id (and optionally guardName).
+     *
+     * @param string|int $permission
+     * @param string|null $guardName
+     *
+     * @return null|\Spatie\Permission\Contracts\Permission
+     */
+    public static function findByNameOrId($permission, $guardName = null)
```

Comment:
What are your thoughts about whether this public method should also be part of the interface? 

I'm okay with leaving it `public` since it can probably be adapted to support uuid/guid (which are strings). But that makes me wonder whether another name might be better for this method.  (If it was protected/private then renaming is more trivial.)
What were your thoughts when coming up with naming here?

Reply:
@drbyte actually i thought about a method that encapsulates the logic of (is_string & is_int ) in HasPermission trait .

and while i was thinking, i found that the logic is, wither to find a permission by  name in case the passed $permission parameter is string , or find a permission by id in case the  passed $permission parameter is integer.

so i named the method with findByNameOrId.

[example of is_string then calling findByName and is_int then calling findById](https://github.com/spatie/laravel-permission/blob/master/src/Traits/HasPermissions.php#L108)

if you have a better suggestion or a better approach i would appreciate it .

Reply:
Okay -- your thinking process was the same as mine. :)

I want to think a bit further about the best naming for the public method, since I'd like to expand to uuid as well.
And then must decide whether the Interface/Contract should require the method also.

Reply:
@drbyte , for  the contract i think it should be required , and for naming am not quite sure , but i think we can make something like this : 

```php
public static function findByColumn($permission, $guardName = null, $column = null)
 {
    if (!$column) {
        $column = is_string($permission) ? 'name' : 'id';
     }

     return static::filterPermissionsBy($column, $permission, $guardName)->first();
 }
```

and also change
```php
protected static function filterPermissionsBy(string $key, $value, string $guardName)

//instead of $key to be $column

protected static function filterPermissionsBy(string $column, $value, string $guardName)
```